api/src/main/java/org/xwiki/apps/googleapps/GoogleAppsScriptService.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.apps.googleapps;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.stability.Unstable;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.drive.model.File;
+import com.google.api.services.drive.model.FileList;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.api.Document;
+import com.xpn.xwiki.api.Object;
+
+/**
+ * Script service containing the methods used by the view files contained in the ui module.
+ *
+ * @since 3.0
+ * @version $Id$
+ */
+@Component
+@Named("googleApps")
+@Singleton
+public class GoogleAppsScriptService implements ScriptService
+{
+
+    @Inject
+    private GoogleAppsManager manager;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    /**
+     * @return if the application is licensed and activated
+     * @throws XWikiException in case a context cannot be read from thread.
+     * @since 3.0
+     */
+    @Unstable
+    public boolean isActive() throws XWikiException {
```

Comment:
Script services don't throw exceptions normally.

Reply:
Done in a5a69382ef5d8946f3b83605b3fd82bd967fb5ec